src/js/tracks/text-track.js

```diff
@@ -262,9 +214,7 @@ class TextTrack extends EventTarget {
       tt.loaded_ = true;
     }
 
-    if (browser.IS_IE8) {
-      return tt;
-    }
+    return retval;
```

Comment:
The way constructors in js work when called with `new` is that a new object is allocated and made available in via `this` but if you return a *different* object, the return value for `new Foo()` will be that different object. We need to do that for IE8 because IE8 doesn't let us use getters (`defineProperty`) on non-HTML elements. So, in IE8, we create a custom element, attach the getters and setters to it and then return that object. Only IE8 has this limitation and videojs 5.x must continue supporting IE8 at least for existing features and new features cannot break IE8, though, we could decide on a case-by-case basis whether new features should support IE8.